PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -534,147 +598,38 @@ struct lambdaCorrelationAnalysis {
     histos.fill(HIST("Event/h1d_posz"), collision.posZ());
     histos.fill(HIST("Event/h1d_ft0m_mult_percentile"), collision.cent());
 
-    int nLaP = 0, nLaM = 0, nLa = 0;
-    std::array<float, 2> lambda_mass = static_cast<std::array<float, 2>>(cfg_lambda_mass);
-    std::array<float, 2> lambda_left = static_cast<std::array<float, 2>>(cfg_lambda_left);
-    std::array<float, 2> lambda_right = static_cast<std::array<float, 2>>(cfg_lambda_right);
+    std::vector<std::vector<float>> v_mass_win = {{cfg_lambda_mass_min, cfg_lambda_mass_max}, {cfg_lambda_left_min, cfg_lambda_left_max}, {cfg_lambda_right_min, cfg_lambda_right_max}};
 
     auto lambda_tracks = part_lambda_tracks->sliceByCached(aod::lambdatrack::lambdaCollisionId, collision.globalIndex(), cache);
     auto anti_lambda_tracks = part_anti_lambda_tracks->sliceByCached(aod::lambdatrack::lambdaCollisionId, collision.globalIndex(), cache);
 
-    // Lambda
     for (auto const& lambda_1 : lambda_tracks) {
-
-      // Lambda Mass Window
-      if (lambda_1.mass() > lambda_mass[0] && lambda_1.mass() < lambda_mass[1]) {
-        fillHistos<0>(lambda_1);
-        partSingle<0, 0>(lambda_1);
-        ++nLaP;
-        ++nLa;
-
-        // lambda-antilambda
-        for (auto const& antilambda_1 : anti_lambda_tracks) {
-          if (antilambda_1.mass() > lambda_mass[0] && antilambda_1.mass() < lambda_mass[1]) {
-            partPair<0, 0>(lambda_1, antilambda_1);
-          }
-        }
-
-        // lambda-lambda
-        for (auto const& lambda_2 : lambda_tracks) {
-          if ((lambda_2.index() != lambda_1.index()) && (lambda_1.postrackid() != lambda_2.postrackid()) && (lambda_1.negtrackid() != lambda_2.negtrackid())) {
-            if (lambda_2.mass() > lambda_mass[0] && lambda_2.mass() < lambda_mass[1]) {
-              partPair<0, 1>(lambda_1, lambda_2);
-            }
-          }
-        }
+      analyseSingles<kLambda, kCentralWindow>(lambda_1, v_mass_win);
+      analyseSingles<kLambda, kLeftWindow>(lambda_1, v_mass_win);
+      analyseSingles<kLambda, kRightWindow>(lambda_1, v_mass_win);
+      for (auto const& antilambda_2 : anti_lambda_tracks) {
+        analysePairs<kLambdaAntiLambda, kCentralWindow, false>(lambda_1, antilambda_2, v_mass_win);
+        analysePairs<kLambdaAntiLambda, kLeftWindow, false>(lambda_1, antilambda_2, v_mass_win);
+        analysePairs<kLambdaAntiLambda, kRightWindow, false>(lambda_1, antilambda_2, v_mass_win);
```

Comment:
As mentioned before this is not efficient at the pair level unless it is only called for pairs that are already within the required acceptances